packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx

```diff
@@ -10,46 +10,69 @@ import {
     Text,
     useCopyToClipboard,
 } from '@deriv/components';
+import { websiteUrl } from '@deriv/shared';
 import { observer } from '@deriv/stores';
 import { Localize, localize } from 'Components/i18next';
 import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
 import MyProfileSeparatorContainer from 'Components/my-profile/my-profile-separator-container';
+import { buy_sell } from 'Constants/buy-sell';
+import { ad_type } from 'Constants/floating-rate';
 import { TAdvert } from 'Types';
-import ShareMyAdsIcons from './share-my-ads-socials';
 import ShareMyAdsCard from './share-my-ads-card';
+import ShareMyAdsSocials from './share-my-ads-socials';
 
 const ShareMyAdsModal = ({ advert }: TAdvert) => {
     const [is_copied, copyToClipboard, setIsCopied] = useCopyToClipboard();
+    const { account_currency, advertiser_details, id, local_currency, rate_display, rate_type, type } = advert;
+    const { id: advertiser_id } = advertiser_details;
+    const { hideModal, is_modal_open } = useModalManagerContext();
 
     const divRef = React.useRef(null);
-    // TODO: replace with proper url when available
-    const advert_url = window.location.href;
-
-    const { hideModal, is_modal_open } = useModalManagerContext();
+    const advert_url = `${websiteUrl()}cashier/p2p/advertiser?id=${advertiser_id}&advert_id=${id}`;
+    const custom_message_values = { account_currency, advert_url, local_currency, rate_display };
+    const getCustomMessage = () => {
+        if (rate_type === ad_type.FLOAT) {
+            return type === buy_sell.BUY
+                ? localize(
+                      "Hi! I'd like to exchange {{local_currency}} for {{account_currency}} at {{rate_display}}% on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
+                      custom_message_values
+                  )
+                : localize(
+                      "Hi! I'd like to exchange {{account_currency}} for {{local_currency}} at {{rate_display}}% on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
+                      custom_message_values
+                  );
+        }
+        return type === buy_sell.BUY
+            ? localize(
+                  "Hi! I'd like to exchange {{local_currency}} for {{account_currency}} at {{rate_display}} {{local_currency}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
+                  custom_message_values
+              )
+            : localize(
+                  "Hi! I'd like to exchange {{account_currency}} for {{local_currency}} at {{rate_display}} {{local_currency}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
+                  custom_message_values
+              );
+    };
```

Comment:
```suggestion
    const getCustomMessage = () => {
        if (rate_type === ad_type.FLOAT) {
            return type === buy_sell.BUY
                ? localize(
                      "Hi! I'd like to exchange {{local_currency}} for {{account_currency}} at {{rate_display}}% on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
                      custom_message_values
                  )
                : localize(
                      "Hi! I'd like to exchange {{account_currency}} for {{local_currency}} at {{rate_display}}% on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
                      custom_message_values
                  );
        }
        return type === buy_sell.BUY
            ? localize(
                  "Hi! I'd like to exchange {{local_currency}} for {{account_currency}} at {{rate_display}} {{local_currency}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
                  custom_message_values
              )
            : localize(
                  "Hi! I'd like to exchange {{account_currency}} for {{local_currency}} at {{rate_display}} {{local_currency}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
                  custom_message_values
              );
    };
```
The base string is the same, only the values differ so a lot of these conditional can be simplified using <Localize /> components via the `values` props.